src/iop/useless.c

```diff
@@ -41,17 +60,24 @@ typedef struct dt_iop_useless_params_t
   // stored in self->params and self->default_params
   // also, since this is stored in db, you should keep changes to this struct
   // to a minimum. if you have to change this struct, it will break
-  // users data bases, and you should increment the version
-  // of DT_MODULE(VERSION) above!
-  int checker_scale;
-  float factor;
+  // users data bases, and you have to increment the version
+  // of DT_MODULE(VERSION) above and provide a legacy_params upgrade path!
+  // Tags in the comments get picked up by the introspection framework and are
+  // used to set range and default for widgets in gui_init, labels (for widgets and history)
+  // and value checks before commit_params.
+  // If no explicit init() is specified, the default implementation uses $DEFAULT tags
+  // to initialise self->default_params, which is then used in gui_init to set widget defaults.
+  int checker_scale; // $MIN: 0 $MAX: 10 $DEFAULT: 1 $DESCRIPTION: "size"
+  float factor;      // $MIN: -5.0 $MAX: 5.0 $DEFAULT: 0
+  gboolean check;    // $DESCRIPTION: "checkbox option"
+  dt_iop_useless_type_t method; // $DEFAULT: DT_USELESS_SECOND $DESCRIPTION: "parameter choices"
```

Comment:
You might want to explicitly state that the above are examples as it's not clear from the names or descriptions (like it is clear for the combobox typedef above)

Reply:
that applies to everything in the whole module though...
"This is an example implementation..."
so if this was going to be repeated here, maybe it could make clear that applies to everything in the rest of the module as well, rather then repeating it everywhere which could become tiresome? I'll take suggestions for a phrase!

Reply:
Thing is some parts of the example module are a general template (should be used in any module) and some are part of the module-specific functionality (the example). It's not always entirely clear which is which, though mostly it is implied by context. Perhaps if you used the same naming convention as you do for the struct itself and call the variables useless_scale, useless_factor etc?

Reply:
I don't _disagree_ with you, and I see how it takes a _little bit_ of effort to deduce from context, but if someone can't manage that, they have no hope of getting any of the rest right and definitely should not start by implementing a module from scratch. It may be a kindness to scare them off at an early stage!

(I'm only half joking here)

Adding useless_ to everything might suggest that the naming convention is to add the module name in front of every field, which I personally would not want to encourage.

Reply:
> useless_ to everything might suggest that the naming convention

I did think of that. I'm a long way from being able to implement a module from scratch so maybe I should be scared off. How about if you prefixed them all with some_ instead of having them be semi-meaningful. So `some_int` `some_float` `some_gboolean`.